core/src/main/java/com/google/cloud/sql/core/InstanceMetadata.java

```diff
@@ -18,24 +18,55 @@
 
 import com.google.cloud.sql.IpType;
 import java.security.cert.Certificate;
+import java.util.List;
 import java.util.Map;
 
 /** Represents the results of @link #fetchMetadata(). */
 class InstanceMetadata {
 
+  private final CloudSqlInstanceName instanceName;
   private final Map<IpType, String> ipAddrs;
-  private final Certificate instanceCaCertificate;
+  private final List<Certificate> instanceCaCertificates;
+  private final boolean casManagedCertificate;
+  private final String dnsName;
+  private final boolean pscEnabled;
 
-  InstanceMetadata(Map<IpType, String> ipAddrs, Certificate instanceCaCertificate) {
+  InstanceMetadata(
+      CloudSqlInstanceName instanceName,
+      Map<IpType, String> ipAddrs,
+      List<Certificate> instanceCaCertificates,
+      boolean casManagedCertificate,
+      String dnsName,
+      boolean pscEnabled) {
+    this.instanceName = instanceName;
     this.ipAddrs = ipAddrs;
-    this.instanceCaCertificate = instanceCaCertificate;
+    this.instanceCaCertificates = instanceCaCertificates;
+    this.casManagedCertificate = casManagedCertificate;
```

Comment:
nit: may be rename this to `isCasManagedCertificate`.

Reply:
Usually we don't prefix private boolean fields with "is" this in java. The accessor method is prefixed with "is". 

```java 
class InstanceMetadata {
private boolean casManagedCertificate
public boolean isCasManagedCertificate() {
  return casManagedCertificate;
}
}
```